src/index.js

```diff
@@ -278,11 +303,20 @@ class FloodSub extends BaseProtocol {
     function checkIfReady (peer) {
       if (peer && peer.isWritable) {
         peer.sendUnsubscriptions(topics)
-      } else {
-        setImmediate(checkIfReady.bind(peer))
```

Comment:
This was doing some recursion on the check, removing this makes that no longer happen. Does this still need to recursively check?